packages/account/src/Components/leave-confirm/__tests__/leave-confirm.spec.js

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import { render, screen, fireEvent, waitFor } from '@testing-library/react';
+import LeaveConfirm, { TransitionBlockerWithRouter } from '../leave-confirm';
+import { Formik } from 'formik';
+import { Router } from 'react-router';
+import { createBrowserHistory } from 'history';
+import { isMobile } from '@deriv/shared';
+
+let modal_root_el;
+
+jest.mock("@deriv/shared", () => ({
+    ...jest.requireActual("@deriv/shared"),
+    isMobile: jest.fn(() => false),
+}));
+
+
+beforeAll(() => {
+    modal_root_el = document.createElement('div');
+    modal_root_el.setAttribute('id', 'modal_root');
+    document.body.appendChild(modal_root_el);
+});
+
+afterAll(() => {
+    document.body.removeChild(modal_root_el);
+});
+
+const LeaveConfirmComponent = () => {
+    const history = createBrowserHistory();
+    return (
+        <Router history={history}>
+            <Formik>
+                <LeaveConfirm />
+            </Formik>
+        </Router>
+    )
+}
+
+const withRouter = (Component) => {
+    const history = createBrowserHistory();
+    return props => {
+        return <Router history={history}>
+            <Component {...props} />
+        </Router>
+
+    }
+}
+
+const TransitionBlockerComponent = withRouter(TransitionBlockerWithRouter);
+
+describe('LeaveConfirm', () => {
+     it('should render LeaveConfirm component in desktop mode', () => {
+        jest.spyOn(React, 'useState').mockReturnValueOnce([true, () => {}])
+        render(<LeaveConfirmComponent />);
+    });
+    it('should render LeaveConfirm component in mobile mode', () => {
+        jest.spyOn(React, 'useState').mockImplementationOnce(() => React.useState(true));
+        isMobile.mockReturnValueOnce(true)
+        render(<LeaveConfirmComponent />);
+    });
+    it("should show proper icon", () => {
+        jest.spyOn(React, 'useState').mockImplementationOnce(() => React.useState(true));
+
+        render(<LeaveConfirmComponent />);
+        expect(screen.getByTestId('unsaved_changes_icon')).toBeInTheDocument();
+        expect(screen.getByText('Unsaved changes')).toBeInTheDocument();
+        expect(screen.getByText('You have unsaved changes. Are you sure you want to discard changes and leave this page?')).toBeInTheDocument();
+        expect(screen.getByRole('button', { name: 'Cancel' })).toBeInTheDocument();
+        expect(screen.getByRole('button', { name: 'Leave Settings' })).toBeInTheDocument();
+    })
+    it("should trigger onclick callback when the user clicks cancel button", async () => {
+        jest.spyOn(React, 'useState').mockImplementationOnce(() => React.useState(true));
+
+        render(<LeaveConfirmComponent />);
+        const el_cancel_btn = screen.getByRole('button', { name: 'Cancel' });
+        fireEvent.click(el_cancel_btn);
+        await waitFor(() => {
+            expect(screen.queryByText('Unsaved changes')).not.toBeInTheDocument();
+        })
```

Comment:
maybe we can check here if something exists in the document after fire event?

Reply:
After this event we need to be sure that modal window is hidden, so this check does opposite thing.